components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/utils/mappings/ApplicationKeyMappingUtil.java

```diff
@@ -81,7 +83,12 @@ public static ApplicationKeyDTO fromApplicationKeyToDTO(APIKey apiKey) {
         applicationKeyDTO.setConsumerKey(apiKey.getConsumerKey());
         applicationKeyDTO.setConsumerSecret(apiKey.getConsumerSecret());
         applicationKeyDTO.setKeyState(apiKey.getState());
-        applicationKeyDTO.setSupportedGrantTypes(null); //this is not supported by impl yet
+        if (apiKey.getGrantTypes() != null) {
+            applicationKeyDTO.setSupportedGrantTypes(Arrays.asList(apiKey.getGrantTypes().split(" ")));
+        } else {
+            applicationKeyDTO.setSupportedGrantTypes(null);
```

Comment:
Do we explicitly have to set null here. It will be null anyway

Reply:
By default it's an empty list. Kept it that way to keep backward compatibility.